proto_generator/protogenerator.go

```diff
@@ -52,6 +52,7 @@ var (
 	preferOperationalState               = flag.Bool("prefer_operational_state", false, "If set to true, state (config false) fields in the YANG schema are preferred over intended config leaves in the generated messages with compressed schema paths. This flag is only valid for compress_paths=true and exclude_state=false.")
 	skipEnumDedup                        = flag.Bool("skip_enum_deduplication", false, "If set to true, all leaves of type enumeration will have a unique enum output for them, rather than sharing a common type (default behaviour).")
 	useDefiningModuleForTypedefEnumNames = flag.Bool("typedef_enum_with_defmod", false, "If set to true, all typedefs of type enumeration or identity will be prefixed with the name of its module of definition instead of its residing module.")
+	useConsistentNamesForProtoUnionEnums = flag.Bool("consistent_union_enum_names", false, "If set to true, uses more consistent names for enumerations defined within a union. This flag is only active when typedef_enum_with_defmod is also set.")
```

Comment:
I've left it out for users for that reason. I still want to keep "proto" because within enumgen.go there is no longer that context (as these all become free-floating flags), and using a different name once the flag gets there makes the code harder to read IMO, which defeats the point of keeping the name tidy.